PyInstaller/utils/win32/winresource.py

```diff
@@ -199,8 +199,7 @@ def UpdateResources(dstpath, data, type_, names=None, languages=None):
             for language in res[type_][name]:
                 logger.info("Updating resource type %s name %s language %s",
                             type_, name, language)
-                win32api.UpdateResource(hdst, type_, name,
-                                        data.encode('UTF-8'), language)
+                win32api.UpdateResource(hdst, type_, name, data, language)
```

Comment:
In #2675 there is a distinction whether data is `str` or `bytes`. Is this necessary or not? Also see https://github.com/mhammond/pywin32/issues/694. Pinging @tallforasmurf who as I assume posted that issue.

Reply:
I've searched through the code and found only one place where `UpdateResources` may be called with `str`: https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/win32/winmanifest.py#L938. Perhaps, data should be encoded here, instead of letting `UpdateResources` accept `str`.

If this seems to be insufficient, I will add type check to `UpdateResources`.

Reply:
Thanks a lot for investigating this. [The line you mentioned](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/PyInstaller/utils/win32/winmanifest.py#L938) calls [`self.toprettyxml()`](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/PyInstaller/utils/win32/winmanifest.py#L904), which [already encodes the string](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/PyInstaller/utils/win32/winmanifest.py#L912). Thus indeed there is no need to add a type check here nor in UpdateResources. I'll merge this later this weekend

Reply:
A [few lines later](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/PyInstaller/utils/win32/winmanifest.py#L915) `xmlstr` gets decoded again. Is this unintentional?

Reply:
@tallforasmurf So you are suggesting to post-bone this patch? You statement is correct, but if nobody will soon review the code as you described, this patch might be open for month if not years. What do you thing?

Reply:
@htgoebel, I will try to find some spare time in the next week to look through the code and check `UpdateResources` calls.

In the meantime, I added a fix for manifest update.